### PR TITLE
Change the type of padding input to array

### LINKED ISSUE #224: Use tensor type for the padding rather than MLOperand
The pad in the [spec](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-pad) uses the `MLOperand padding` as its sencond input. But I find that the pad in [DML](https://docs.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_padding_operator_desc) and in [IE](https://docs.openvino.ai./latest/openvino_docs_ops_movement_Pad_1.html), padding was defined with a tensor type. I think it's better to use tensor type to avoid the gap between spec and implementation.

### PR DESCRIPTION
**Pull request by huningxin**
fix #224

@wchao1115 @pyu10055 @anssiko , PTAL.

***
<a href="https://pr-preview.s3.amazonaws.com/huningxin/webnn/pull/233.html" title="Last updated on Nov 24, 2021, 12:27 AM UTC (379149e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/233/aaddda1...huningxin:379149e.html" title="Last updated on Nov 24, 2021, 12:27 AM UTC (379149e)">Diff</a>

### CODE REVIEW

**Comment by dontcallmedom** [comment]
I don't know why it didn't run nor how to find out why; I guess further commits might trigger it (but I'm not sure)

**Comment by anssiko** [comment]
The padding argument has semantics that cannot be fully expressed in terms of Web IDL (`sequence<sequence<long>>`) so we need the prose as in this PR to define what is a valid argument and what's not. This is one example where using algorithmic prose might improve readability (tracked in https://github.com/webmachinelearning/webnn/issues/210).

To take a concrete example, if a malformed padding argument is provided, do we expect `compute()` to throw a DataError DOMException? Do we want to fail sooner than that? Failing immediately upon invocation of `pad()` would be helpful for eager execution I believe.

Not a blocker for this PR, but a comment for future enhancement.

**Review by anssiko — APPROVED** [review]
Unblocking review with a comment on future enhancement.

**Comment by dontcallmedom** [comment]
> The padding argument has semantics that cannot be fully expressed in terms of Web IDL (`sequence<sequence<long>>`) 

As far as I can tell, this is a valid WebIDL construct.

**Comment by huningxin** [comment]
> Can we trigger it?

Retriggered. Now the preview is available.

> To take a concrete example, if a malformed padding argument is provided, do we expect `compute()` to throw a DataError DOMException? Do we want to fail sooner than that? Failing immediately upon invocation of `pad()` would be helpful for eager execution I believe.

+1 to fail upon invocation of `pad()`.

> This is one example where using algorithmic prose might improve readability (tracked in #210).

+1. I'll look into it and probably address it in a separate PR. Thanks @anssiko .

**Comment by anssiko** [comment]
@huningxin I opened https://github.com/webmachinelearning/webnn/issues/234 so that "fail upon invocation" can be addressed separately. I think also some other ops may be affected.

**Review by anssiko — APPROVED** [review]

**Comment by huningxin** [comment]
a soft ping @wchao1115 @pyu10055 , thanks!

File: index.bs [hunk 1]
```diff
@@ -1563,13 +1563,14 @@ dictionary MLPadOptions {
 };
 
 partial interface MLGraphBuilder {
-  MLOperand pad(MLOperand input, MLOperand padding, optional MLPadOptions options = {});
+  MLOperand pad(MLOperand input, sequence<sequence<long>> padding,
+                optional MLPadOptions options = {});
```
Comment (wchao1115) [inline index.bs:1567]: My only holdback on this is that the `padding` data itself may be unknown during graph construction time, and that having a placeholder operand will allow for the dynamic padding at an execution time. ONNX for example has this provision for the pad op.

Comment (huningxin) [reply index.bs:1567]: According to the implementation feedback (#224), the major issue is that native ML APIs, like DirectML ([`DML_PADDING_OPERATOR_DESC`](https://docs.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_padding_operator_desc)), only support static padding data.

Comment (wchao1115) [reply index.bs:1567]: This is actually a shortcoming in DirectML, which is being fixed.

Comment (huningxin) [reply index.bs:1567]: Thanks @wchao1115 for the inputs. I have passed your feedback to the original issue and proposed to keep the current version. Once we make consensus there, I'll close this PR.

**Comment by huningxin** [comment]
Per feedback of https://github.com/webmachinelearning/webnn/issues/224#issuecomment-1033397241, I'll close this PR and keep the current definition.

**Pull request closed without merging**